rfc/text/advanced/ap_10_rpc_progressive_calls.md

```diff
@@ -1,5 +1,230 @@
+   |                   |---------------------->|
+   |                   |                       |
+   | CALL (progress)   |                       |
+   |------------------>|                       |
+   |                   | INVOCATION (progress) |
+   |                   |---------------------->|
+   |                   | YIELD (progress)      |
+   |                   |<----------------------|
+   |                   |                       |
+   | RESULT (progress) |                       |
+   |<------------------|                       |
+   |                   |                       |
+   |                   | YIELD (progress)      |
+   |                   |<----------------------|
+   |                   |                       |
+   | RESULT (progress) |                       |
+   |<------------------|                       |
+   |                   |                       |
+   |       ...         |       ...             |
+   |                   |                       |
+   |  CALL (final)     |                       |
+   |------------------>|                       |
+   |                   | INVOCATION (final)    |
+   |                   |---------------------->|
+   |                   |                       |
+   |                   |  YIELD or ERROR       |
+   |                   |<----------------------|
+   |                   |                       |
+   |  RESULT or ERROR  |                       |
+   |<------------------|                       |
+,--+---.            ,--+---.                ,--+---.
+|Caller|            |Dealer|                |Callee|
+`------'            `------'                `------'
+
+A *Caller* indicates its willingness to issue a progressive call by setting
+
+{align="left"}
+CALL.Options.progress|bool := true
+
+*Example.* Caller-to-Dealer `CALL`
+
+{align="left"}
+```json
+    [
+        48,
+        77245,
+        {
+            "progress": true
+        },
+        "com.myapp.get_country_by_coords",
+        [50.450001, 30.523333]
+    ]
+```
+
+If the *Callee* supports progressive calls, the *Dealer* will forward the *Caller's* willingness to send progressive calls by setting
+
+{align="left"}
+INVOCATION.Details.progress|bool := true
+
+
+*Example.* Dealer-to-Callee `INVOCATION`
+
+{align="left"}
+```json
+    [
+        68,
+        35224,
+        379,
+        {
+            "progress": true
+        },
+        [50.450001, 30.523333]
+    ]
+```
+
+A call invocation MUST *always* end in a *normal* `CALL` without `"progress": true` option, or explicitly set `"progress": false` which is default.
+
+**Progressive calls and shared registration**
+
+RPCs can have a multiple registrations (see `Shared Registration` feature) with different `<invocation_policies>`.
+In this case `progressive` `CALLs` can be routed to different *Callees*, what can lead to unexpected results.
+To bind `INVOCATIONs` to the same *Callee* *Caller* can specify `sticky` option during `CALL`
```

Comment:
This is a very powerful feature. Practically relevant, and very hard to do manually. Very good!